iOS/Projects/Features/Record/Sources/Presentation/WorkoutEnvironmentScene/ViewModel/WorkoutEnvironmentSetupViewModel.swift

```diff
@@ -147,9 +147,28 @@ extension WorkoutEnvironmentSetupViewModel: WorkoutEnvironmentSetupViewModelRepr
     return Publishers.MergeMany(workoutTypes, idle, workoutPeerType, didSelectWorkoutType, didSelectWorkoutPeerType).eraseToAnyPublisher()
   }
 
-  func didTapStartButton() {
-    let workoutSettiong: WorkoutSetting = .init(mode: .run, environment: .multi, opponents: [])
-    coordinator?.pushPeerRandomMatchingViewController(workoutSetting: workoutSettiong)
+  private func didTapStartButton() {
+    guard
+      let didSelectWorkoutType,
+      let didSelectWorkoutPeerType,
+      let mode = WorkoutPeerTypeCodeToModeString(rawValue: didSelectWorkoutPeerType.typeCode)
+    else {
+      return
+    }
+
```

Comment:
p3: guard-else문에는 Log를 넣어주는 것도 좋을 것 같아요. Log를 작성하는 습관을 들이면 디버깅할 때 편하더라고요 ㅎㅎ

```suggestion
      let mode = WorkoutPeerTypeCodeToModeString(rawValue: didSelectWorkoutPeerType.typeCode)
    else {
      Log.make().error("....")
      return
    }

```

Reply:
저도 이부분 고민 했는데요, 현재 Log가 print처럼 어느 부분에서 에러가 나는지 직관적으로 확인할 수 있는 것 같습니다. 그런데 문제점이라면 많은 곳에서 Log를 남용할 경우 관리가 되지 않는 다는 문제점이 발생할 것 같아서, Log사용했던 대부분의 코드를 지우고 올렸습니다. 혹시 Log와 관련되어서 코드에 직접적으로 남기는 것이 괜찮으신지 궁금합니다! 만약 괜찮으시다면 이것에 관한 규칙을 정했으면 좋겠습니다.

Reply:
저는 guard 구문은 웬만하면 넣는 편이에요. 안그러면 오류가 어디서 났는지 파악하기 어렵고, guard에 막혔다는 건 예외라는 뜻이기도 하니까요. 그만큼 많은 로그가 나타나지 않을 거라 생각해요.
그리고 로그도 `debug`, `info`, `notice`, `error`, `fault`여서 심각성에 따라서 필터링해서 로그를 확인할 수 있는 시스템이 마련되어있습니다. 그래서 되도록 넣는 게 좋지 않을까 싶어요. :)